cassiopeia/simulator/TreeSimulator.py

```diff
@@ -0,0 +1,19 @@
+"""
+Abstract class TreeSimulator, for tree simulation module.
+
+All tree simulators are derived classes of this abstract class, and at a minimum
+implement a method called `simulate_tree`.
+"""
+import abc
+from cassiopeia.data import CassiopeiaTree
+
+
+class TreeSimulator(abc.ABC):
+    """
+    TreeSimulator is an abstract class that all tree simulators derive from.
+    """
```

Comment:
Can you define what a `TreeSimulator` is? If this is just a topology, we should make it clear. What is this intended to return? How are users supposed to interface with it? How would someone extend it?

Reply:
Thanks, I've added a more lengthy description!